### PR TITLE
Compat update from RUM data

### PR DESCRIPTION
**Pull request by triblondon**
This PR updates targeting files based on results from compatibility tests performed on sites that opt-in to our real user monitoring program. Notably, many features we polyfill are now available in iOS 10.

Also updated the RumReport to require at least 20 unique IPs and 3 unique referrers for a result to count.  In future I'd suggest we raise this further as we get more participation in the RUM program.

(seems Github doesn't support markdown tables in `<details>` blocks)

| feature_name | ua_family | ua_version | pass_rate_pc | ip_count | refer_source_count | total_count | is_targeted | targeting_status |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Array.of | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| Array.of | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| Array.of | safari | 9.1.0 | 100 | 119 | 4 | 311 | Yes | False positive - remove targeting |
| CustomEvent | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| DOMTokenList | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| Element.prototype.after | chrome | 55.0.0 | 100 | 54 | 4 | 155 | Yes | False positive - remove targeting |
| Element.prototype.after | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| Element.prototype.after | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| Element.prototype.append | chrome | 55.0.0 | 100 | 54 | 4 | 155 | Yes | False positive - remove targeting |
| Element.prototype.append | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| Element.prototype.append | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| Element.prototype.before | chrome | 55.0.0 | 100 | 54 | 4 | 155 | Yes | False positive - remove targeting |
| Element.prototype.before | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| Element.prototype.before | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| Element.prototype.closest | ios_saf | 9.0.0 | 100 | 229 | 6 | 664 | Yes | False positive - remove targeting |
| Element.prototype.prepend | chrome | 55.0.0 | 100 | 54 | 4 | 155 | Yes | False positive - remove targeting |
| Element.prototype.prepend | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| Element.prototype.prepend | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| Element.prototype.remove | firefox | 45.0.0 | 100 | 22 | 4 | 40 | Yes | False positive - remove targeting |
| Element.prototype.remove | firefox | 48.0.0 | 100 | 112 | 5 | 332 | Yes | False positive - remove targeting |
| Element.prototype.remove | firefox | 49.0.0 | 100 | 305 | 8 | 864 | Yes | False positive - remove targeting |
| Element.prototype.remove | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| Element.prototype.remove | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| Element.prototype.remove | ios_saf | 9.0.0 | 100 | 229 | 6 | 664 | Yes | False positive - remove targeting |
| Element.prototype.remove | ios_saf | 9.3.0 | 100 | 188 | 4 | 272 | Yes | False positive - remove targeting |
| Element.prototype.remove | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| Element.prototype.remove | safari | 9.1.0 | 100 | 119 | 4 | 311 | Yes | False positive - remove targeting |
| Element.prototype.replaceWith | chrome | 55.0.0 | 100 | 54 | 4 | 155 | Yes | False positive - remove targeting |
| Element.prototype.replaceWith | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| Element.prototype.replaceWith | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| Node.prototype.contains | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| Number.isNaN | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| Number.isNaN | ios_saf | 9.0.0 | 100 | 229 | 6 | 664 | Yes | False positive - remove targeting |
| Object.assign | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| Object.assign | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| Object.assign | ios_saf | 9.0.0 | 100 | 229 | 6 | 664 | Yes | False positive - remove targeting |
| Object.assign | ios_saf | 9.3.0 | 100 | 188 | 4 | 272 | Yes | False positive - remove targeting |
| String.prototype.endsWith | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| String.prototype.endsWith | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| String.prototype.endsWith | ios_saf | 9.0.0 | 100 | 229 | 6 | 664 | Yes | False positive - remove targeting |
| String.prototype.endsWith | ios_saf | 9.3.0 | 100 | 188 | 4 | 272 | Yes | False positive - remove targeting |
| String.prototype.endsWith | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| String.prototype.endsWith | safari | 9.1.0 | 100 | 119 | 4 | 311 | Yes | False positive - remove targeting |
| String.prototype.includes | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| String.prototype.includes | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| String.prototype.includes | ios_saf | 9.0.0 | 100 | 229 | 6 | 664 | Yes | False positive - remove targeting |
| String.prototype.includes | ios_saf | 9.3.0 | 100 | 188 | 4 | 272 | Yes | False positive - remove targeting |
| String.prototype.includes | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| String.prototype.includes | safari | 9.1.0 | 100 | 119 | 4 | 311 | Yes | False positive - remove targeting |
| String.prototype.startsWith | ie | 14.14393.0 | 100 | 53 | 4 | 102 | Yes | False positive - remove targeting |
| String.prototype.startsWith | ios_saf | 10.0.0 | 100 | 325 | 4 | 559 | Yes | False positive - remove targeting |
| String.prototype.startsWith | ios_saf | 9.0.0 | 100 | 229 | 6 | 664 | Yes | False positive - remove targeting |
| String.prototype.startsWith | ios_saf | 9.3.0 | 100 | 188 | 4 | 272 | Yes | False positive - remove targeting |
| String.prototype.startsWith | safari | 10.0.0 | 100 | 150 | 6 | 472 | Yes | False positive - remove targeting |
| String.prototype.startsWith | safari | 9.1.0 | 100 | 119 | 4 | 311 | Yes | False positive - remove targeting |
| ~html5-elements | ie | 9.0.0 | 100 | 21 | 4 | 25 | Yes | False positive - remove targeting |

### CODE REVIEW

**Comment by JakeChampion** [comment]
This is so good! Great work!

File: service/RumReport.js [hunk 1]
```diff
@@ -141,16 +142,12 @@ function Compat() {
...
-								rec.targeting_status = 'False positive';
-							} else if (!isTargeted && rec.pass_rate < 20) {
-								rec.targeting_status = 'False negative';
+							if (isTargeted && rec.pass_rate_pc > 80) {
+								rec.targeting_status = 'False positive - remove targeting';
+							} else if (!isTargeted && rec.pass_rate_pc < 20) {
+								rec.targeting_status = 'False negative - add targeting';
 							} else {
-								rec.targeting_status = 'Inconclusive';
+								return null; // Correctly targeted
```
Comment (JakeChampion) [inline service/RumReport.js:150]: 👍

File: polyfills/String/prototype/startsWith/config.json [hunk 2]
```diff
@@ -14,10 +14,10 @@
 		"chrome": "<=40",
 		"firefox": "3.6 - 17",
 		"opera": "10 - 18",
-		"safari": "*",
-		"ie": "6 - *",
+		"safari": "<9",
```
Comment (JakeChampion) [inline polyfills/String/prototype/startsWith/config.json:17]: I see safari 9.1 in the table, not safari 9, are we sure on this change?

**Review by JakeChampion — CHANGES_REQUESTED** [review]
I don't see a change made for ~html5-elements to remove ie9 from targeting even though the table specifies we should.

File: polyfills/String/prototype/startsWith/config.json (hunk 2 above)
Comment (triblondon) [reply polyfills/String/prototype/startsWith/config.json:17]: No, I admit I'm slightly extrapolating from the data for `endsWith`

**Comment by triblondon** [comment]
I didn't change html5-elements because I suspect the detection of HTML5 styling support is not perfect science. Feel free to check whether html5shivoffers any benefits in ie9 and if not we can definitely remove it.

**Review by JakeChampion — APPROVED** [review]

**Pull request merged**